ZelBack/src/services/appsService.js

```diff
@@ -8871,6 +8871,93 @@ async function rescanGlobalAppsInformationAPI(req, res) {
   }
 }
 
+/**
+ * Check if we are misssing over 95% of the app hashes and in that case, get from one of the peers the permanentappmessages and process them if it was messages received
+ */
+let checkAndSyncAppHashesWasEverExecuted = false;
+async function checkAndSyncAppHashes() {
+  try {
+    const dbopen = dbHelper.databaseConnection();
+    const database = dbopen.db(config.database.daemon.database);
+    // get flux app hashes that do not have a message;
+    const query = { };
+    const projection = {
+      projection: {
+        _id: 0,
+        message: 1,
+      },
+    };
+    const results = await dbHelper.findInDatabase(database, appsHashesCollection, query, projection);
+    const numberOfMissingApps = results.filter((app) => app.message === false).length;
+    if (numberOfMissingApps > results.length * 0.95) {
+      let finished = false;
+      let i = 0;
+      while (!finished && i <= 5) {
+        i += 1;
+        const client = outgoingPeers[Math.floor(Math.random() * outgoingPeers.length)];
+        let axiosConfig = {
+          timeout: 5000,
+        };
+        log.info(`checkAndSyncAppHashes - Getting explorer sync status from ${client.ip}:${client.port}`);
+        // eslint-disable-next-line no-await-in-loop
+        const response = await serviceHelper.axiosGet(`http://${client.ip}:${client.port}/explorer/sync`, axiosConfig).catch((error) => log.error(error));
```

Comment:
likely should be the new endpoint `issynced` instead of `sync`

Reply:
yeah I ahve changed after the name of the router and forgot to update it on the code. Nice catch, thanks. It's fixed